jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -204,17 +204,24 @@ public PathResource(Path path)
      * @param parent the parent path resource
      * @param childPath the child sub path
      */
-    private PathResource(PathResource parent, String childPath) throws MalformedURLException
+    private PathResource(PathResource parent, String childPath)
     {
         // Calculate the URI and the path separately, so that any aliasing done by
-        // FileSystem.getPath(path,childPath) is visiable as a difference to the URI
+        // FileSystem.getPath(path,childPath) is visible as a difference to the URI
         // obtained via URIUtil.addDecodedPath(uri,childPath)
 
-        this.path = parent.path.getFileSystem().getPath(parent.path.toString(), childPath);
-        if (isDirectory() &&!childPath.endsWith("/"))
-            childPath+="/";
-        this.uri = URIUtil.addPath(parent.uri,childPath);
-        this.alias = checkAliasPath();
+        try
+        {
+            this.path = parent.path.getFileSystem().getPath(parent.path.toString(), childPath);
+            if (isDirectory() && !childPath.endsWith("/"))
+                childPath += "/";
+            this.uri = URIUtil.addPath(parent.uri, childPath);
+            this.alias = checkAliasPath();
+        }
+        catch(InvalidPathException e)
+        {
+            throw (InvalidPathException) new InvalidPathException(childPath, e.getReason()).initCause(e);
```

Comment:
why the coerce?

Reply:
InvalidPathException includes details in its message about the entire filesystem path.  This trims the exception to only reference the child segment being "Added to"

Reply:
The `initCause()` returns Throwable